### PR TITLE
#1652 Run OpenRewrite to migrate langchain4j-core to java 17

### PR DESCRIPTION
**Pull request by kpavlov**
## Issue

Contributes to #1652

## Change

Run OpenRewrite [recipe](https://docs.openrewrite.org/running-recipes/popular-recipe-guides/migrate-to-java-17) to migrate core module to Java 17.

This pull request focuses on modernizing the codebase by incorporating Java's pattern matching for `instanceof` and replacing deprecated methods with newer alternatives.

## General checklist

- [x] There are no breaking changes
- [ ] I have added unit and integration tests for my change
- [x] I have manually run all the unit tests, and they are all green
- [ ] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [ ] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

## Checklist for adding new maven module

- [ ] I have added my new module in the root `pom.xml` and `langchain4j-bom/pom.xml`

## Checklist for adding new embedding store integration

- [ ] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreRemovalIT` that extends from `EmbeddingStoreWithRemovalIT`

## Checklist for changing existing embedding store integration

- [ ] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

### CODE REVIEW

File: langchain4j-core/src/main/java/dev/langchain4j/rag/content/injector/DefaultContentInjector.java [hunk 1]
```diff
@@ -46,10 +46,12 @@
...
     public static final PromptTemplate DEFAULT_PROMPT_TEMPLATE = PromptTemplate.from(
-            "{{userMessage}}\n" +
-                    "\n" +
-                    "Answer using the following information:\n" +
-                    "{{contents}}"
+            """
+            {{userMessage}}
+            
+            Answer using the following information:
+            {{contents}}\
```
Comment (langchain4j) [inline langchain4j-core/src/main/java/dev/langchain4j/rag/content/injector/DefaultContentInjector.java]:
```suggestion
            {{contents}}
```
Is `\` needed?

File: langchain4j-core/src/main/java/dev/langchain4j/rag/query/router/LanguageModelQueryRouter.java [hunk 2]
```diff
@@ -45,12 +45,14 @@ public class LanguageModelQueryRouter implements QueryRouter {
...
-                    "It is very important that your answer consists of either a single number " +
-                    "or multiple numbers separated by commas and nothing else!\n" +
-                    "User query: {{query}}"
+            """
+            Based on the user query, determine the most suitable data source(s) \
+            to retrieve relevant information from the following options:
+            {{options}}
+            It is very important that your answer consists of either a single number \
+            or multiple numbers separated by commas and nothing else!
+            User query: {{query}}\
```
Comment (langchain4j) [inline langchain4j-core/src/main/java/dev/langchain4j/rag/query/router/LanguageModelQueryRouter.java]:
```suggestion
            User query: {{query}}
```

File: langchain4j-core/src/main/java/dev/langchain4j/rag/query/transformer/ExpandingQueryTransformer.java [hunk 3]
```diff
@@ -36,13 +36,15 @@
...
-                    "without enumerations, hyphens, or any additional formatting!\n" +
-                    "User query: {{query}}"
+            """
+            Generate {{n}} different versions of a provided user query. \
+            Each version should be worded differently, using synonyms or alternative sentence structures, \
+            but they should all retain the original meaning. \
+            These versions will be used to retrieve relevant documents. \
+            It is very important to provide each query version on a separate line, \
+            without enumerations, hyphens, or any additional formatting!
+            User query: {{query}}\
```
Comment (langchain4j) [inline langchain4j-core/src/main/java/dev/langchain4j/rag/query/transformer/ExpandingQueryTransformer.java]:
```suggestion
            User query: {{query}}
```

File: langchain4j-core/src/test/java/dev/langchain4j/internal/JsonTest.java [hunk 4]
```diff
@@ -28,11 +28,13 @@ void conversionToJsonAndFromJsonWorks() {
...
-        "  \"sampleDate\": \"2023-01-15\",\n" +
-        "  \"sampleDateTime\": \"2023-01-15T10:20:00\",\n" +
-        "  \"some_value\": \"value\"\n" +
-        "}"
+        """
+        {
+          "sampleDate": "2023-01-15",
+          "sampleDateTime": "2023-01-15T10:20:00",
+          "some_value": "value"
+        }\
```
Comment (langchain4j) [inline langchain4j-core/src/test/java/dev/langchain4j/internal/JsonTest.java]:
```suggestion
        }
```

**Review by ghost — COMMENTED** [review]
@kpavlov thank you!

**Comment by kpavlov** [comment]
Everything should be good now

**Review by ghost — APPROVED** [review]
@kpavlov thank you!

**Pull request merged**